code_samples/discounts/src/Command/OrderPriceCommand.php

```diff
@@ -0,0 +1,134 @@
+        $orderIdentifier = '4315bc58-1e96-4f21-82a0-15f736cbc4bc';
+        $currencyCode = 'EUR';
+
+        $output->writeln('Product data:');
+        $product = $this->productService->getProduct($productCode);
+        $currency = $this->currencyService->getCurrencyByCode($currencyCode);
+
+        $basePrice = $this->productPriceService->getPriceByProductAndCurrency($product, $currency);
+        $resolvedPrice = $this->priceResolver->resolvePrice($product, new PriceContext($currency));
+
+        if ($resolvedPrice === null) {
+            throw new Exception('Could not resolve price for the product');
+        }
+
+        $output->writeln(sprintf('Base price: %s', $this->formatPrice($basePrice->getMoney())));
+        $output->writeln(sprintf('Discounted price: %s', $this->formatPrice($resolvedPrice->getMoney())));
+
+        if ($resolvedPrice instanceof PriceEnvelopeInterface) {
+            /** @var \Ibexa\Discounts\Value\Price\Stamp\DiscountStamp $discountStamp */
+            foreach ($resolvedPrice->all(DiscountStamp::class) as $discountStamp) {
+                $output->writeln(
+                    sprintf(
+                        'Discount applied: %s , new amount: %s',
+                        $discountStamp->getDiscount()->getName(),
+                        $this->formatPrice(
+                            $discountStamp->getNewPrice()
+                        )
+                    )
+                );
+            }
+        }
+
+        $output->writeln('Order details:');
+
+        $order = $this->orderService->getOrderByIdentifier($orderIdentifier);
+        foreach ($order->getItems() as $item) {
+            /** @var ?DiscountsData $discountData */
+            $discountData = $item->getContext()['discount_data'] ?? null;
+            if ($discountData instanceof DiscountsData) {
+                $output->writeln(
+                    sprintf(
+                        'Product bought with discount: %s, base price: %s, discounted price: %s',
+                        $item->getProduct()->getName(),
+                        $this->formatPrice($discountData->getOriginalPrice()),
+                        $this->formatPrice(
+                            $item->getValue()->getUnitPriceGross()
+                        )
+                    )
+                );
+            } else {
+                $output->writeln(
+                    sprintf(
+                        'Product bought with original price: %s, price: %s',
+                        $item->getProduct()->getName(),
+                        $this->formatPrice(
+                            $item->getValue()->getUnitPriceGross()
+                        )
+                    )
+                );
+            }
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function formatPrice(Money $money): string
+    {
+        return $money->getAmount() / 100.0 . ' ' . $money->getCurrency()->getCode();
```

Comment:
100 isn't correct for all currencies. It depends on the currency's subunit.

I wonder if `$money->getCurrency()` returns a https://doc.ibexa.co/en/5.0/api/php_api/php_api_reference/classes/Ibexa-Contracts-ProductCatalog-Values-CurrencyInterface.html
If yes, here is the fix:

```suggestion
        return $money->getAmount() / pow(10, $money->getCurrency()->getSubUnits()) . ' ' . $money->getCurrency()->getCode();
```